akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -723,4 +730,73 @@ final class ProcessorModulePhase(materializer: PhasedFusingActorMaterializer) ex
   override def takePublisher(slot: Int, publisher: Publisher[Any]): Unit = publisher.subscribe(processor)
 
   override def onIslandReady(): Unit = ()
+}
+
+object TlsModuleIslandTag extends IslandTag
+
+final class TlsModulePhase(materializer: PhasedFusingActorMaterializer) extends PhaseIsland[TlsModulePhase.MaterializedInfo] {
+  import TlsModulePhase._
+
+  def name: String = "TlsModulePhase"
+
+  def materializeAtomic(mod: AtomicModule[Shape, Any], attributes: Attributes): (MaterializedInfo, Any) = {
+    val es = materializer.effectiveSettings(attributes)
+    println(mod)
+    val tls = mod.asInstanceOf[TlsModule]
+
+    val props = TLSActor.props(materializer.settings, tls.createSSLEngine, tls.verifySession, tls.closing)
+    val impl = materializer.actorOf(props, s"tls-${nextId()}", es.dispatcher)
+    def factory(id: Int) = new ActorPublisher[Any](impl) {
+      override val wakeUpMsg = FanOut.SubstreamSubscribePending(id)
+    }
+    val publishers = Vector.tabulate(2)(factory)
+    impl ! FanOut.ExposedPublishers(publishers)
+
+    /*    assignPort(tls.plainOut, publishers(TLSActor.UserOut))
+    assignPort(tls.cipherOut, publishers(TLSActor.TransportOut))
+
+    assignPort(tls.plainIn, FanIn.SubInput[Any](impl, TLSActor.UserIn))
+    assignPort(tls.cipherIn, FanIn.SubInput[Any](impl, TLSActor.TransportIn))*/
+
+    //matVal.put(atomic, NotUsed)
+    (MaterializedInfo(
+      tls,
+      impl,
+      Seq(
+        tls.plainOut → publishers(TLSActor.UserOut),
+        tls.cipherOut → publishers(TLSActor.TransportOut)
+      )
+    ), NotUsed)
+  }
+  def assignPort(in: InPort, slot: Int, logic: MaterializedInfo): Unit = logic.assignPort(in, slot)
+  def assignPort(out: OutPort, slot: Int, logic: MaterializedInfo): Unit = ()
+
+  def createPublisher(out: OutPort, logic: MaterializedInfo): Publisher[Any] =
+    logic.outPorts.find(_._1 == out).get._2
+
+  protected def takePublisher(slot: Int, publisher: Publisher[Any]): Unit = ???
+
+  override def takePublisher( /*in: InPort, */ slot: Int, publisher: Publisher[Any], logic: MaterializedInfo): Unit = {
+    val in = logic.slots.find(_._1 == slot).get._2
```

Comment:
no need, there are only two slots here, the two inputs. A simple `if` statement is enough as slot can be either 0 or 1 and it is deterministic which input port it means (the first or second).